Annotations/annotation_generator.py

```diff
@@ -0,0 +1,392 @@
+#! python3 annotation_generator.py <ast txt file> <output for annotations file> 
+# generate AST text with: clang -Xclang -ast-dump -fsyntax-only -fno-color-diagnostics <c file> > <ast txt file>
+
```

Comment:
This code is complex enough that you should add some tests for it. A few shell scripts that execute this script on the tests (using the command lines below) and compare its results to the expected output should be sufficient; just make sure that they run in CI.